### PR TITLE
Add the known alts panel, which prevents suspicious login logs for whitelisted ckeys

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Adds a way to track "known alts". If a ckey connects with a banned known alt in connection history, it will not log it as suspicious. This only works if BOTH ckeys match. If john's known alt is john2, and john3 connects with john2 in history, it will still log. 

## Why It's Good For The Game
Players will often times switch to new accounts, and admins will perma-ban the old one to make sure they stick on the same account. However, this ends up cluttering suspicious_logins and chat logs with these alt accounts. Some admins prefer to make watchlist entries, but we'd really like to just have them banned.

## Changelog

:cl:
admin: Adds a panel for "known alts", which let you write down confirmed, valid alts between two people so that they don't show in suspicious connection history.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
I don't see a problem with this, I can apply the SQL tomorrow.

File: code/modules/admin/known_alts.dm [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+				load_known_alts()
+
+			qdel(query_delete_known_alt)
+			show_panel(usr.client)
+
+/// Returns the list of known alts, will return an empty list if the DB could not be connected to.
+/// This proc can block.
+/datum/known_alts/proc/load_known_alts()
+	if (!isnull(cached_known_alts))
+		return cached_known_alts
```
Comment (MrStonedOne) [inline code/modules/admin/known_alts.dm:126]: Cached sql data has to be timed and have a ttl.

If the ttl is expired, but it fails to fetch the data, it should then return the stale cached data.

See the stickyban subsystem for further details.

https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#sql:

>The ttl for data from the database is 10 seconds. You must have a compelling reason to store and reuse data for longer then this.

File: code/modules/admin/known_alts.dm [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+			query_already_exists.warn_execute()
+
+			if (query_already_exists.last_error)
+				qdel(query_already_exists)
+				return
+
+			if (query_already_exists.NextRow())
+				alert(usr, "Those two are already in the list of known alts!")
+				return
```
Comment (Jordie0608) [inline code/modules/admin/known_alts.dm:52]: Need to qdel query here before return.

File: code/modules/admin/known_alts.dm [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+				"id" = id,
+			))
+
+			if (!query_known_alt_info.warn_execute())
+				qdel(query_known_alt_info)
+				return
+
+			if (!query_known_alt_info.NextRow())
+				alert("Couldn't find the known alt with the ID [id]")
+				return
```
Comment (Jordie0608) [inline code/modules/admin/known_alts.dm:98]: Need to qdel query here before return.

File: code/modules/admin/known_alts.dm [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+			var/list/result = query_known_alt_info.item
+
```
Comment (Jordie0608) [inline code/modules/admin/known_alts.dm:101]: And qdel the query once it's used.

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

**Pull request merged**